store-failures

```diff
@@ -128,8 +128,19 @@ def insert_failure(cursor, run_id, testname, retry_reason):
 
 
 def find_failed_tests(fp):
+    last_msg = ""
+    save_message = False
     for line in fp:
         line = line.decode('utf-8')
+
+        if save_message:
+            last_msg = line
+            save_message = False
+            continue
+
+        if line.startswith("testlib.Error: FAIL: Test completed, but found unexpected journal messages"):
```

Comment:
This is way too much "magic string".  The chance of this message being casually changed in the future is pretty high, and it's likely they wouldn't also change it here.

why not just store some extra lines in all cases?

Reply:
> This is way too much "magic string". The chance of this message being casually changed in the future is pretty high, and it's likely they wouldn't also change it here.

yeah... maybe we can set up test for it?

> why not just store some extra lines in all cases?

How do you mean?

Reply:
meh `test/common/run-tests` in cockpit has this line:

`test/common/run-tests:    unexpected_message = b"Test completed, but found unexpected" in test.output` 

So I'm fine with this as is tbh. The way to avoid this would be to have that as a constant somewhere in bots, like error types or something? But not sure if that's worth the trouble.

Reply:
Right, but that at least avoids the precise exception prefix ("testlib."), and the precise composition of the error message, and just restricts itself to the single string. I'd feel a bit better with doing it the same way. It's already hackish enough ;-) With shortening the string I'm ok with this. Thanks!

Reply:
The only issue is that there are two lines containing this string:
```
    raise Error("FAIL: Test completed, but found unexpected journal messages:\n" + first)
testlib.Error: FAIL: Test completed, but found unexpected journal messages:
Failed to generate stack trace: invalid `Elf' handle
```

we would most likely need to match for that space before "FAIL". Or edit test/common but that is slow to propagate to other projects.

Reply:
I think this has me won over as well then. And with these hacks I'd rather have it be as specific of a string as possible :)